### PR TITLE
TST: add workflow_dispatch action to test pysatNASA with latest pysat RC

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses ROADMAP

Adds an additional workflow to download pysat from test.pypi for integration testing with future pysat releases.  Workflow is manually triggered (button will appear when merged into main).  Pushing this forward now since we are about to relase new code.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Setting to push and inspecting the logs at https://github.com/pysat/pysatNASA/actions/runs/3346690449 to ensure that the correct things are installed.

## Test Configuration
* Operating system: Mac and Windows
* Version number: Python 3.10
* Github Actions

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

**Comment by jklenzing** [comment]
Rationale:
- Just using "latest" python version supported.  This will only run from main, so updates will be infrequent in theory.  Enforcing NEP29 here will only cause more headaches.
- Testing for both ubuntu and windows
- We could add tests with the pysatCDF RC as well.  This could either be in a single workflow (easier to maintain, more jobs to run each time) or as two separate workflows.

@aburrell? @rstoneback? Thoughts?

File: .github/workflows/pysat_rc.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      run: |
+        mkdir pysatData
+        python -c "import pysat; pysat.params['data_dirs'] = 'pysatData'"
+
+    - name: Test PEP8 compliance
+      run: flake8 . --count --select=D,E,F,H,W --show-source --statistics
+
+    - name: Evaluate complexity
+      run: flake8 . --count --exit-zero --max-complexity=10 --statistics
+
```
Comment (jklenzing) [inline .github/workflows/pysat_rc.yml:44]: probably could drop pep8 since this is handled elsewhere

File: .github/workflows/pysat_rc.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+name: Test with latest pysat RC
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (aburrell) [inline .github/workflows/pysat_rc.yml:13]: Include mac as well

File: .github/workflows/pysat_rc.yml (hunk 1 above)
Comment (aburrell) [reply .github/workflows/pysat_rc.yml:44]: This is so short, I am ok either way.

**Comment by jklenzing** [comment]
Having had a little more time to think about, I'm inclined to leave pysatCDF RC testing out for the time being.  The no-binary option complicates things.

**Comment by jklenzing** [comment]
Added mac tests for both workflows.  Needed a gfortran reinstall to get pysatCDF to work.

**Comment by jklenzing** [comment]
Still need a review here before the RC is finalized

**Review by JonathonMSmith — APPROVED** [review]

**Comment by jklenzing** [comment]
Thanks @JonathonMSmith!

**Pull request merged**